src/plots/font_attributes.js

```diff
@@ -86,6 +86,42 @@ module.exports = function(opts) {
             ].join(' ')
         },
 
+        capitalize: opts.noFontCapitalize ? undefined : {
+            editType: editType,
+            valType: 'enumerated',
+            values: ['none', 'word', 'uppercase', 'lowercase'],
+            dflt: 'none',
+            description: [
+                'Sets capitalization of text.',
+                'It can be used to make text appear in all-uppercase or all-lowercase,',
+                'or with each word capitalized.'
+            ].join(' ')
+        },
+
+        striding: opts.noFontStriding ? undefined : {
+            editType: editType,
+            valType: 'flaglist',
+            flags: ['under', 'over', 'through'],
+            extras: ['none'],
+            dflt: 'none',
+            description: [
+                'Sets the kind of decoration line(s) with text,',
+                'such as an *under*, *over* or *through*',
+                'as well as combinations e.g. *under+over*, etc.'
```

Comment:
`text-decoration` has [so many options](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration) I kind of think we should just call the attribute `decoration`, leave it as an unstructured string, and point users to the reference so if they want `'underline overline wavy red'` they can have it.

Reply:
Not all those options are working as SVG attribute.
Here is another link: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/text-decoration

Reply:
I'm on the fence -- I think `decoration` is a much better API, and it's more future-proof for if/when SVG supports more decoration options.

But at the same time it's potentially frustrating for users if we name it the same as the CSS property but only support a fraction of the options. We can state clearly in the docs which properties are supported, but not everyone will read the docs carefully.

 I wonder how likely it is that SVG will support additional decoration options in the near-mid future.

Reply:
This option is now named `decorline`. After lots of testing (including Chrome version used in Kaleido), I don't see we could/should expose more options now.
Alternatively we may rename this attribute to `underline` as boolean which is the most useful option (IMHO) and also allow other options as flag list string.

Reply:
Also another option might be to call it `lineposition` (or `linepos`)? 
This might be better as we declare the position of decoration line(s).

Reply:
How do you feel about naming it `lined`?
```
lined: 'under'
```

Reply:
How about `textline`?

Reply:
`textline`, `line`, as well as `decorline` are not favorites because they may point to the lines drawn around each letter.

Reply:
Ok, this one is tricky to name but out of the suggested options I think I like `lineposition` best so far.

Reply:
Applied in [a344464](https://github.com/plotly/plotly.js/pull/6983/commits/a34446425f8aa9dcef40968c5fb03b0694b56d54).